docs/deploy/ios.rst

```diff
@@ -176,6 +176,257 @@ This built-in list can be controlled by editing ``prepare_params.sh``.
+
+.. code:: shell
+
+    # Create directory
+    mkdir -p dist/models && cd dist/models
+    # Clone HF weights
+    git lfs install
+    git clone https://huggingface.co/mlabonne/NeuralHermes-2.5-Mistral-7B
+    cd ../..
+    # Convert weight
+    mlc_chat convert_weight ./dist/models/NeuralHermes-2.5-Mistral-7B/ \
+        --quantization q4f16_1 \
+        -o dist/NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC
+
+**Step 2 Generate MLC Chat Config**
+
+Use ``mlc_chat gen_config`` to generate ``mlc-chat-config.json`` and process tokenizers.
+See :ref:`compile-command-specification` for specification of ``gen_config``.
+
+.. code:: shell
+
+    mlc_chat gen_config ./dist/models/NeuralHermes-2.5-Mistral-7B/ \
+        --quantization q3f16_1 --conv-template neural_hermes_mistral \
+        -o dist/NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC
+
+For the ``conv-template``, `conv_template.cc <https://github.com/mlc-ai/mlc-llm/blob/main/cpp/conv_templates.cc>`__
+contains a full list of conversation templates that MLC provides.
+
+If the model you are adding requires a new conversation template, you would need to add your own. 
+Follow `this PR <https://github.com/mlc-ai/mlc-llm/pull/1402>`__ as an example. 
+We look up the template to use with the ``conv_template`` field in ``mlc-chat-config.json``.
+
+For more details, please see :ref:`configure-mlc-chat-json`.
+
+**Step 3 Upload weights to HF**
+
+.. code:: shell
+
+    # First, please create a repository on Hugging Face.
+    # With the repository created, run
+    git lfs install
+    git clone https://huggingface.co/my-huggingface-account/my-mistral-weight-huggingface-repo
+    cd my-mistral-weight-huggingface-repo
+    cp path/to/mlc-llm/dist/NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC/* .
+    git add . && git commit -m "Add mistral model weights"
+    git push origin main
+
+After successfully following all steps, you should end up with a Huggingface repo similar to 
+`NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC <https://huggingface.co/mlc-ai/NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC>`__,
+which includes the converted/quantized weights, the ``mlc-chat-config.json``, and tokenizer files.
+
+
+**Step 4 Register as a ModelRecord**
+
+Finally, we modify the code snippet for
+`app-config.json <https://github.com/mlc-ai/mlc-llm/blob/main/ios/MLCChat/app-config.json>`__
+pasted above.
+
+We simply specify the Huggingface link as ``model_url``, while reusing the ``model_lib`` for 
+``Mistral-7B``.
+
+.. code:: javascript
+   
+   "model_list": [
+      // Other records here omitted...
+      {
+         // Substitute model_url with the one you created `my-huggingface-account/my-mistral-weight-huggingface-repo`
+         "model_url": "https://huggingface.co/mlc-ai/NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC",
+         "model_id": "Mistral-7B-Instruct-v0.2-q3f16_1",
+         "model_lib": "mistral_q3f16_1",
+         "model_lib_path": "lib/Mistral-7B-Instruct-v0.2/Mistral-7B-Instruct-v0.2-q3f16_1-iphone.tar",
```

Comment:
is model_lib_path necessary? I feel this is snot something we need during runtime